server/src/test/java/com/broadcom/lsp/cobol/core/model/SearchVariableTest.java

```diff
@@ -0,0 +1,111 @@
+import com.broadcom.lsp.cobol.core.model.variables.Variable;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/** This test check functions of the {@link SearchVariable}. */
+class SearchVariableTest {
+
+  private static final Locality LOCALITY = Locality.builder().build();
+
+  @Test
+  void empty() {
+    SearchVariable searchVariable = new SearchVariable(List.of());
+    assertTrue(searchVariable.findVariables("foo", List.of()).isEmpty());
+  }
+
+  @Test
+  void findByNameWithAnyParent() {
+    List<Variable> variables = collectVariables(
+        var("FOO"),
+        var("BAR",
+            var("FOO",
+                var("BAZ")),
+            var("BAZ",
+                var("FOO")))
+    );
+    List<Variable> found = new SearchVariable(variables).findVariables("FOO", List.of());
+    checkQualifiers(found, "FOO", "BAR FOO", "BAR BAZ FOO");
+  }
+
+  @Test
+  void findByNameAndParent() {
+    List<Variable> variables = collectVariables(
+        var("FOO"),
+        var("BAR",
+            var("FOO",
+                var("BAZ")),
+            var("BAZ",
+                var("FOO")))
+    );
+    List<Variable> found = new SearchVariable(variables).findVariables("FOO", List.of("BAR"));
+    checkQualifiers(found, "BAR FOO", "BAR BAZ FOO");
+  }
+
+  @Test
+  void findByNameAndTwoParents() {
+    List<Variable> variables = collectVariables(
+        var("FOO",
+            var("BAZ",
+                var("BAR"),
+                var("QWE",
+                    var("BAR"))),
+            var("QWE",
+                var("BAZ",
+                    var("BAR")))),
+        var("BAZ",
+            var("FOO",
+                var("BAR")))
+    );
+    List<Variable> found = new SearchVariable(variables).findVariables("BAR", List.of("BAZ", "FOO"));
+    checkQualifiers(found, "FOO BAZ BAR", "FOO BAZ QWE BAR", "FOO QWE BAZ BAR");
+  }
+
+  private static void checkQualifiers(List<Variable> variables, String... qualifiers) {
+    assertEquals(Set.of(qualifiers), variables.stream().map(Variable::getQualifier).collect(Collectors.toSet()));
+  }
+
+  private static List<Variable> collectVariables(Function<Variable, List<Variable>>... factories) {
+    List<Variable> result = new ArrayList<>();
+    for (var factory: factories)
+      result.addAll(factory.apply(null));
+    return result;
+  }
+
+  private static Function<Variable, List<Variable>> var(String name, Function<Variable, List<Variable>>... factories) {
+    return (Variable parent) -> {
+      String qualifier = Optional.ofNullable(parent)
+          .map(it -> it.getQualifier() + " ")
+          .orElse("") + name;
+      List<Variable> result = new ArrayList<>();
+      Variable it = new ElementItem(name, qualifier, LOCALITY, parent, "", "", null);
```

Comment:
Locality.builder().build()

Reply:
In this case, it will create a new instance on each call without the necessity

Reply:
I don't think that it will be an issue for this test.